tests/system/Database/Builder/WhereTest.php

```diff
@@ -446,4 +449,119 @@ public function testWhereWithLower()
         $expectedSQL = 'SELECT * FROM "jobs" WHERE LOWER(jobs.name) = \'accountant\'';
         $this->assertSame($expectedSQL, str_replace("\n", ' ', $builder->getCompiledSelect()));
     }
+
+    public function testWhereValueIsString()
+    {
+        $builder = $this->db->table('users');
+
+        $builder->where('id', '1');
+
+        $expectedSQL = <<<'SQL'
+            SELECT * FROM "users" WHERE "id" = '1'
+            SQL;
+        $this->assertSame($expectedSQL, str_replace("\n", ' ', $builder->getCompiledSelect()));
+    }
+
+    public function testWhereValueIsFloat()
+    {
+        $builder = $this->db->table('users');
+
+        $builder->where('id', 1.234);
+
+        $expectedSQL = <<<'SQL'
+            SELECT * FROM "users" WHERE "id" = 1.234
+            SQL;
+        $this->assertSame($expectedSQL, str_replace("\n", ' ', $builder->getCompiledSelect()));
+    }
+
+    public function testWhereValueIsTrue()
+    {
+        $builder = $this->db->table('users');
+
+        $builder->where('id', true);
+
+        $expectedSQL = 'SELECT * FROM "users" WHERE "id" = 1';
+        $this->assertSame($expectedSQL, str_replace("\n", ' ', $builder->getCompiledSelect()));
+    }
+
+    public function testWhereValueIsFalse()
+    {
+        $builder = $this->db->table('users');
+
+        $builder->where('id', false);
+
+        $expectedSQL = 'SELECT * FROM "users" WHERE "id" = 0';
+        $this->assertSame($expectedSQL, str_replace("\n", ' ', $builder->getCompiledSelect()));
+    }
+
+    public function testWhereValueIsArray()
+    {
+        $builder = $this->db->table('users');
+
+        $builder->where('id', ['a', 'b']);
+
+        // SQL syntax error
+        $expectedSQL = <<<'SQL'
+            SELECT * FROM "users" WHERE "id" = ('a','b')
```

Comment:
Is this the expected result?

Reply:
Pretty much yes. Although we shouldn't provide an array here. The query will have no much sense.